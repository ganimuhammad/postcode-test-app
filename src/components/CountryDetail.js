import React from "react";

function CountryDetail(props) {
  return (
    <div className="ui cards">
      <div className="card">
        <div className="content">
          <div className="header">{props.data.postcode}</div>
          <div className="meta">{props.data.country}</div>
          <div className="description">{props.data.region}</div>
        </div>
      </div>
    </div>
  );
}

export default CountryDetail;
