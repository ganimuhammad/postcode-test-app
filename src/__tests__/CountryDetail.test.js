import React from "react";
import CountryDetail from "../components/CountryDetail";

import { render } from "@testing-library/react";

const stubbedCountry = {
  postcode: "SE18 3AD",
  country: "England",
  region: "London",
};

it("matches snapshot", () => {
  const { asFragment } = render(<CountryDetail data={stubbedCountry} />);

  expect(asFragment()).toMatchSnapshot();
});
