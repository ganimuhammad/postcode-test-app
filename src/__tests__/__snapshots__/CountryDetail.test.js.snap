// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui cards"
  >
    <div
      class="card"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          SE18 3AD
        </div>
        <div
          class="meta"
        >
          England
        </div>
        <div
          class="description"
        >
          London
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
